ssl/statem/extensions_clnt.c

```diff
@@ -538,7 +544,7 @@ static int add_key_share(SSL *s, WPACKET *pkt, unsigned int curve_id)
         assert(s->hello_retry_request);
         if (!s->hello_retry_request) {
             SSLerr(SSL_F_ADD_KEY_SHARE, ERR_R_INTERNAL_ERROR);
-            return 0;
+            return EXT_RETURN_FAIL;
```

Comment:
You said "various" but I could only find this one instance (add_key_share()). Are there others? Changing this one was a mistake I think. It should have just been left as a simple boolean 0 for fail, 1 for success return type. Fix in #3488.